### ISSUE TITLE
The code seems to use the `I18n::locale()` if the param `lang` is empty, but the ConfigValidator doesn't allow us to set an empty string

### ISSUE THREAD

**Issue by Xety** [cakephp-fr/recaptcha#2]
The code seems to use the `I18n::locale()` https://github.com/cakephp-fr/recaptcha/blob/master/src/View/Helper/RecaptchaHelper.php#L64-L67 if the param `lang` is empty (btw it's really a good idea), but the `ConfigValidator` doesn't like when we don't specify any `lang` option : https://github.com/cakephp-fr/recaptcha/blob/master/src/Validation/ConfigValidator.php#L93-L98

ALso, maybe set the `lang` param to be empty by default or to use the `I18n::locale()`? https://github.com/cakephp-fr/recaptcha/blob/master/src/View/Helper/RecaptchaHelper.php#L40

Also, we probably should do a `substr(I18n::locale(), 0, 2))` to get the `I18n::locale()` value because the locale can be also full named such as `fr_FR`, `en_US` etc.

**Comment by cake17** [comment]
@Xety you're right, maybe we could define the lang to be `I18n::locale()` by default.
If no accepted lang is provided (with `I18n::locale` or through the helper), do we define the lang to `en`, or do we throw an exception?

I started to reorder the ConfigValidation and divide it into 2 separate validation:
- one globally for the Plugin : we could use it in bootstrap to check `secret` and all other params 'lang', 'type', ...
- one for the recaptcha helper : only the params 'lang', 'type', ...

Also maybe the `recaptcha.php` config file is not needed, and we could just add the config to `app.php` ?

**Comment by Xety** [comment]
For the lang option like `en-GB`, yeah like you said, the best solution would be to transform them into `en_GB`, so we will follow the CakePHP conventions :
> Translation folders can either be the two letter ISO code of the language or the full locale name such as `fr_FR`, `es_AR`, `da_DK` which contains both the language and the country where it is spoken.

Well, actually, it would be the inverse, let me show you steps by steps how to check it :
`I18n::locale()` result : (These tests should be the same for the `lang` option)
* `en_GB` :
  * 1. `en_GB` has a `_`
  * 2. Transform `en_GB` into `en-GB`
  * 3. Matching `en-GB` in the allowed array

* `en-GB`
  * 1. `en-GB` has a `-`
  * 2. Matching `en-GB` in the allowed array

* `en`
  * 1. `en` has **not** a `_` or a `-`
  * 2. Matching `en` in the allowed array


* `fr_FR` :
  * 1. `fr_FR` has a `_`
  * 2. Transform `fr_FR` into `fr-FR`
  * 3. **Not** matching `fr-FR` in the allowed array
  * 4. Doing a `substr` => `fr`
  * 5. Matching `fr` in the allowed array

* `fr-FR` :
  * 1. `fr-FR` has a `-`
  * 2. **Not** matching `fr-FR` in the allowed array
  * 3. Doing a `substr` => `fr`
  * 4. Matching `fr` in the allowed array

* `fr`
  * 1. `fr` has **not** a `_`
  * 2. Matching `fr` in the allowed array

* `zz`
  * 1. `zz` has **not** a `_`
  * 2. **Not** matching `zz` in the allowed array
  * 3. Use the default `en`

For the `fr-FR`, it's like you want, we can do a `substr` like in my example, or we can refusing it and use the `en`. It's like you want.

For the default locale, i think your solution is the best solution. For me, it would be better to use `en` as default locale if the `I18n::locale()` & the `lang` option has a bad locale.

**Comment by anhtuank7c** [comment]
You all should try this plugin: [recaptcha](https://github.com/crabstudio/recaptcha)

You can define your own `lang` or just leave blank to use your default application locale
